### PR TITLE
GEOT-5442: fix same separator in gml:coordinates

### PR DESCRIPTION
**Pull request by mtoonen**

### CODE REVIEW

File: modules/extension/xsd/xsd-gml2/src/main/java/org/geotools/gml2/bindings/GMLCoordinatesTypeBinding.java [hunk 1]
```diff
@@ -131,13 +131,23 @@ public Object parse(ElementInstance instance, Node node, Object value)
         CoordinateSequence seq = null;
         int i = 0;
         int ncoords = tuples.countTokens(); //number of coordinates
+        if(cs.equals(ts)) {
+            ncoords = ncoords/2;
```
Comment (aaime) [inline modules/extension/xsd/xsd-gml2/src/main/java/org/geotools/gml2/bindings/GMLCoordinatesTypeBinding.java]: Doesn't this assume 2d coordinates? What if the GML has ts=cs but it's representing 3d coordinates?
Hmm... I guess we cannot recognize anymore that case by just looking at the string, like it's done today, one would have to read the srs and make assumptions (it's not uncommon to declare a 2d SRS but provide 3d ordinates).
Oh well, it's not perfect but GML 2 is old and this patch seems to make the parser work better than before for this particular case.

**Comment by aaime** [comment]
Merging, let's see how the CITE tests in GeoServer like this change :-)

File: modules/extension/xsd/xsd-gml2/src/main/java/org/geotools/gml2/bindings/GMLCoordinatesTypeBinding.java (hunk 1 above)
Comment (mtoonen) [inline modules/extension/xsd/xsd-gml2/src/main/java/org/geotools/gml2/bindings/GMLCoordinatesTypeBinding.java]: Yes, you're correct, it does assume 2D, but I figured: it was better than before:-).
However, I might be wrong, but in my tests I think I encountered GML3 geometries, which were parsed with this binding.

**Comment by mtoonen** [comment]
Thanks for reviewing/merging!

**Pull request merged**